core/chat/events/nameChangeEvent.go

```diff
@@ -0,0 +1,26 @@
+package events
+
+// NameChangeEvent is received when a user changes their chat display name.
+type NameChangeEvent struct {
+	Event
+	UserEvent
+	NewName string `json:"newName"`
+}
+
+// NameChangeEventBroadcast is fired when a user changes their chat display name.
+type NameChangeBroadcast struct {
+	Event
+	UserEvent
+	Oldname string `json:"oldName"`
+}
+
+// GetBroadcastPayload will return the object to send to all chat users.
+func (e *NameChangeBroadcast) GetBroadcastPayload() EventPayload {
+	return EventPayload{
+		"id":        e.Id,
+		"timestamp": e.Timestamp,
+		"user":      e.User,
```

Comment:
I did this on purpose so the outbound payloads were explicit and obvious and are not bound by an abstraction, since these values are specifically for client consumption.  I like being able to look at a specific event and know exactly how it's being generated, and it makes it clear when adding more in the future.